src/which.js

```diff
@@ -54,46 +59,35 @@ function _which(options, cmd) {
 
       if (common.platform === 'win') {
         attempt = attempt.toUpperCase();
+      }
 
-        // In case the PATHEXT variable is somehow not set (e.g.
-        // child_process.spawn with an empty environment), use the XP default.
-        var pathExtEnv = process.env.PATHEXT || XP_DEFAULT_PATHEXT;
-        var pathExtArray = splitPath(pathExtEnv.toUpperCase());
-        var i;
-
-        // If the extension is already in PATHEXT, just return that.
-        for (i = 0; i < pathExtArray.length; i++) {
-          var ext = pathExtArray[i];
-          if (attempt.slice(-ext.length) === ext && checkPath(attempt)) {
-            where = attempt;
-            return;
-          }
+      // Cycle through the PATHEXT variable
+      pathExtArray.forEach(function (ext) {
+        // If the argument passed in already has an extension in PATHEXT, just
+        // return that
+        // TODO(nate): replace with .endsWith(ext) once we use v4+
+        if (ext && attempt.slice(-ext.length) === ext && checkPath(attempt)) {
+          where = attempt;
+          return;
         }
 
-        // Cycle through the PATHEXT variable
-        var baseAttempt = attempt;
-        for (i = 0; i < pathExtArray.length; i++) {
-          attempt = baseAttempt + pathExtArray[i];
-          if (checkPath(attempt)) {
-            where = attempt;
-            return;
-          }
-        }
-      } else {
-        // Assume it's Unix-like
-        if (checkPath(attempt)) {
-          where = attempt;
+        // Otherwise, see if it exists without the extension
+        var newAttempt = attempt + ext;
+        if (checkPath(newAttempt)) {
+          where = newAttempt;
           return;
         }
-      }
+      });
     });
   }
 
-  // Command not found anywhere?
-  if (!checkPath(cmd) && !where) return null;
+  // No relative path was specified, so we return what we found from the search
+  if (where) return where;
 
-  where = where || path.resolve(cmd);
+  // Otherwise, return the absolute path of the path specified
+  if (checkPath(cmd)) return path.resolve(cmd);
 
-  return where;
+  // Command not found anywhere?
+  return null;
```

Comment:
`null` is better here because it won't get wrapped by ShellString. The empty string would get wrapped, and `ShellString('')` is actually truthy, unlike `''`.